druid/custom/rich_input/rich_input.lua

```diff
@@ -92,4 +172,55 @@ function RichInput.set_placeholder(self, placeholder_text)
 end
 
 
+function RichInput.on_input(self, action_id, action)
+	self.action_pos_x = action.screen_x
```

Comment:
The values `action_pos_x` and `touch_pos_x` should not be stored in rich input component. You can use it as a local variables inside only on_input functions